src/cascadia/TerminalSettingsModel/ColorScheme.cpp

```diff
@@ -100,7 +100,25 @@ bool ColorScheme::_layerJson(const Json::Value& json)
     // Required fields
     for (unsigned int i = 0; i < TableColors.size(); ++i)
     {
-        isValid &= JsonUtils::GetValueForKey(json, til::at(TableColors, i), til::at(_table, i));
+        std::string_view tableColorName = til::at(TableColors, i);
+        bool valueForKeyIsValid = JsonUtils::GetValueForKey(json, tableColorName, til::at(_table, i));
+
+        // If GetValueForKey failed, try again with alternate color names
+        if (!valueForKeyIsValid)
+        {
+            if (tableColorName == "purple")
+            {
+                tableColorName = "magenta";
+            }
+            else if (tableColorName == "brightPurple")
+            {
+                tableColorName = "brightMagenta";
+            }
+
+            valueForKeyIsValid = JsonUtils::GetValueForKey(json, tableColorName, til::at(_table, i));
+        }
+
+        isValid &= valueForKeyIsValid;

```

Comment:
BTW the second lookup is only necessary if the `tableColorName` value changed.

It's technically possible to solve this issue without additional `if` conditions, by extending our code in its declarative style. Mapping arrays are pretty cool after all! 😄

```cpp
static constexpr std::array<std::pair<std::string_view, size_t>, 18> ColorTableMapping{{
    // Main mappings which are the ones written to settings.json
    {"black", 0},
    {"red", 1},
    {"green", 2},
    {"yellow", 3},
    {"blue", 4},
    {"purple", 5},
    {"cyan", 6},
    {"white", 7},
    {"brightBlack", 8},
    {"brightRed", 9},
    {"brightGreen", 10},
    {"brightYellow", 11},
    {"brightBlue", 12},
    {"brightPurple", 13},
    {"brightCyan", 14},
    {"brightWhite", 15},

    // Extra fallback mappings during reading due to GH#11456
    {"magenta", 5},
    {"brightMagenta", 13},
}};

bool ColorScheme::_layerJson(const Json::Value& json)
{
    // ...

    // Required fields
    size_t colorCount = 0;
    for (const auto& [key, index] : ColorTableMapping)
    {
        colorCount += JsonUtils::GetValueForKey(json, key, til::at(_table, index));
    }
    // We're still valid if we read exactly 16 colors.
    isValid &= colorCount == 16;

    return isValid;
}

Json::Value ColorScheme::ToJson() const
{
    // ...

    for (size_t i = 0; i < 16; ++i)
    {
        const auto& key = til::at(ColorTableMapping, i).first;
        JsonUtils::SetValueForKey(json, key, til::at(_table, i));
    }

    return json;
}
```

Reply:
(this is elegant, I love it)